### PR TITLE
[Quantization] Support Quark int4-fp8 w4a8 for MoE

### PR DESCRIPTION
**Pull request by BowenBao**
This PR extends support of Quark quantized model for int4-fp8 w4a8 quantization spec.

Specifically:
- Weights are double quantized: first at INT4 per channel, then further at FP8 per tensor.
- Activations are quantized dynamically in FP8 per tensor.

We observe large performance uplift and near full accuracy with Kimi-K2-Thinking quantized by Quark with int4-fp8 w4a8 on MI300x.

### CODE REVIEW

File: vllm/model_executor/layers/quantization/quark/quark_moe.py [hunk 1]
```diff
@@ -412,6 +413,183 @@ def apply(
...
+        # GEMM, and weight_scale post
+        for expert_id in range(layer.local_num_experts):
+            layer.w13_weight_scale_2[expert_id] *= max_w13_scales[expert_id]
+            layer.w2_weight_scale_2[expert_id] *= layer.w2_weight_scale[expert_id]
+
+    def get_fused_moe_quant_config(self, layer):
+        return fp8_w8a8_moe_quant_config(
+            w1_scale=layer.w13_weight_scale_2,
+            w2_scale=layer.w2_weight_scale_2,
+            per_out_ch_quant=True,
```
[the remark this answers is not in the document]
Comment (BowenBao) [reply vllm/model_executor/layers/quantization/quark/quark_moe.py:524]: this is handled by the kernel

[the remark this answers is not in the document]
Comment (tjtanaa) [reply vllm/model_executor/layers/quantization/quark/quark_moe.py:524]: @BowenBao May I know which AITER commits is required to support this feature? Does the AITER commit used in the `docker/Dockerfile.rocm_base` support this? 

Is there any weights or quark quantization scripts that we can use to get the quantized model?

[the remark this answers is not in the document]
Comment (BowenBao) [reply vllm/model_executor/layers/quantization/quark/quark_moe.py:524]: hi @tjtanaa, the current AITER commit supports this, it does not require new AITER commit, because the kernel is actually not new.

The quantized model is uploaded to https://huggingface.co/amd/models and will be released at a later time.

**Comment by BowenBao** [comment]
@tjtanaa please take another look, thanks!

File: vllm/model_executor/layers/quantization/quark/quark_moe.py [hunk 2]
```diff
@@ -412,6 +413,183 @@ def apply(
...
+        )
+
+    def apply(
+        self,
+        layer: torch.nn.Module,
```
Comment (tjtanaa) [inline vllm/model_executor/layers/quantization/quark/quark_moe.py]: @BowenBao NITS: I think should this be `layer: FusedMoE` like the other class.

**Review by tjtanaa — APPROVED** [review]
LGTM. Just one part about the type hinting seems offed.
